File: app/src/main/java/dk/aschoen/beatplanner/MetroApp.java
//
// Translated by CS2J (http://www.cs2j.com): 2014-07-29 22:02:12
//

package dk.aschoen.beatplanner;

import android.app.Application;
import android.content.Context;
import android.media.AudioManager;
import android.media.SoundPool;
import android.util.Log;

import dk.aschoen.beatplanner.core.Metronome;

public class MetroApp extends Application
{

    @Override
    public void onCreate() {
        super.onCreate();
        Log.d(getClass().getSimpleName(), "MetroApp starting");
//        Console.WriteLine("MetroApp starting");
    }

    public static Metronome defaultMetronome(Context context) {
        Metronome metro = new Metronome();
        final SoundPool sp = new SoundPool(2, AudioManager.STREAM_MUSIC, 0);
        final int claveId = sp.load(context, R.raw.llfclave, 1);
        final int claveLowId = sp.load(context, R.raw.llfclave_low, 1);

        metro.OnBeatEvent(new Metronome.BeatEventHandler() {
            @Override
            public void handle(int index, int beats, int bars) {
                int sid;
                if (index == 1)
                    sid = claveId;
                else
                    sid = claveLowId;
                sp.play(sid, 1.0f, 1.0f, 1, 0, 1.0f);
            }
        });

        return metro;
    }

}

File: BeatPlannerAndroid/src/beatplanner/ui/MetroApp.cs
using System;
using Android.App;
using Android.Util;
using Android.Runtime;
using BeatPlanner;
using Android.Media;
using Android.Content;

namespace BeatPlannerAndroid
{
  [Application]
  public class MetroApp : Application
  {

    public MetroApp(IntPtr javaReference, JniHandleOwnership transfer) : base(javaReference, transfer)
    {
    }

    public override void OnCreate()
    {
      base.OnCreate();
      Log.Debug(Class.SimpleName, "MetroApp starting");
      Console.WriteLine("MetroApp starting");
    }

    public static Metronome DefaultMetronome(Context context)
    {

      Metronome metro = new Metronome();
      SoundPool sp = new SoundPool(2, Android.Media.Stream.Music, 0);
      int claveId = sp.Load(context, Resource.Raw.llfclave, 1);
      int claveLowId = sp.Load(context, Resource.Raw.llfclave_low, 1);
      metro.BeatEvent += (object sender, BeatEventArgs e) =>
      {
        int sid;
        if (e.Index == 1)
          sid = claveId;
        else
          sid = claveLowId;
        sp.Play(sid, 1.0f, 1.0f, 1, 0, 1.0f);
      };

      return metro;
    }



  }
}

